src/lib/HasuraClient.ts

```diff
@@ -11,13 +12,32 @@ const authLink = setContext((_, { headers }) => {
   return {
     headers: {
       ...headers,
-      authorization: token ? `Bearer ${token}` : "",
+      /* 
+        We send a value here ("Bearer ") even when token is 
+        missing because Hasura interprets missing authorization 
+        headers and even empty string as a public user.
+
+        Instead we want Hasura to return an "invalid-jwt" error
+        so that we can check for it as a condition to refresh
+        the token.
+      */
+      authorization: `Bearer ${token}`,
     },
   };
 });
 
+const errorLink = onError(({ graphQLErrors }) => {
+  if (graphQLErrors) {
+    graphQLErrors.map(({ extensions }) => {
+      if (extensions?.code === "invalid-jwt") {
+        AuthenticationService.refreshHasuraToken();
```

Comment:
The logic to retry the query after refreshing the token will be addressed in future PRs.

As well as handling network errors.